activity.go

```diff
@@ -92,11 +122,53 @@ func (w *activityCtxWrapper) Name() string {
 }
 
 func (w *activityCtxWrapper) GetInput(name string) interface{} {
-	return w.ctx.GetInput(name)
+
+	val := w.ctx.GetInput(name)
+
+	// if the input value is complex, we need to modify it
+	if oldMdInput := w.legacyAct.Metadata().Input; oldMdInput != nil {
+		if attr, ok := oldMdInput[name]; ok {
+			if attr.Type() == legacyData.TypeComplexObject {
+				md := ""
+
+				//see if there is a corresponding input schema to construct complex object
+				if sIO, ok := w.ctx.(schema.HasSchemaIO); ok {
+					s := sIO.GetInputSchema(name)
```

Comment:
s := sIO.GetInputSchema(name) returns nil. I think because we never handle/save complex object's metadata(schema) that in app.json. such as:
```
              "input": {
                "Method": "GET",
                "Uri": "https://petstore.swagger.io/v2/pet/4",
                "requestType": "application/json",
                "responseType": "application/json",
                "queryParams": {
                  "metadata": "{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"string\"}},\"required\":[]}",
                  "value": ""
                },
                "headers": {
                  "metadata": "{\"type\":\"object\",\"properties\":{\"Accept\":{\"type\":\"string\"},\"Accept-Charset\":{\"type\":\"string\"},\"Accept-Encoding\":{\"type\":\"string\"},\"Content-Type\":{\"type\":\"string\"},\"Content-Length\":{\"type\":\"string\"},\"Connection\":{\"type\":\"string\"},\"Cookie\":{\"type\":\"string\"},\"Pragma\":{\"type\":\"string\"}},\"required\":[]}",
                  "value": ""
                }
              },
              "output": {
                "responseBody": {
                  "metadata": "{\"$schema\":\"http://json-schema.org/draft-04/schema#\",\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"number\"},\"category\":{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"number\"},\"name\":{\"type\":\"string\"}}},\"name\":{\"type\":\"string\"},\"photoUrls\":{\"type\":\"array\",\"items\":{\"type\":\"string\"}},\"tags\":{\"type\":\"array\",\"items\":{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"number\"},\"name\":{\"type\":\"string\"}}}},\"status\":{\"type\":\"string\"}}}",
                  "value": ""
                },
                "headers": {
                  "metadata": "{\"type\":\"object\",\"properties\":{\"Accept\":{\"type\":\"string\"},\"Accept-Charset\":{\"type\":\"string\"},\"Accept-Encoding\":{\"type\":\"string\"},\"Content-Type\":{\"type\":\"string\"},\"Content-Length\":{\"type\":\"string\"},\"Connection\":{\"type\":\"string\"},\"Cookie\":{\"type\":\"string\"},\"Pragma\":{\"type\":\"string\"}},\"required\":[]}",
                  "value": ""
                }
              },
```

Reply:
Currently the code only support the use of the new explicit schema support and converts objects with schemas to compex objects on the legacy side.  I will add a conversion of complex objects in json to the new schema format separately